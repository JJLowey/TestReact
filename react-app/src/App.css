body, #root
{
    font-family: "Open Sans",Helvetica,Arial,sans-serif!important;
    color: #444;
    height: 100vh;
    background-color: #0b76c5;
}

.container
{
    padding-top: 10vh;
    z-index: 15px;
}

.searchContainer
{
    max-width: 475px;
    background-color: #f3ce56;
    border-radius: 4px;
    padding: 20px;
    margin: auto;
    margin-top: 40px;
}

.searchContainer h1
{
    font-size: 24px!important;
    padding-bottom: 5px;
    font-weight: 700px;
}

#resultsContainer
{
    list-style-type: none;
    padding: 5px;
    margin-bottom: 0;
}

#resultsContainer li
{
    margin-top: 5px;
}

.title
{
    font-weight: 700!important;
}

.sub-title
{
    font-weight: 500;
}

#pickUpInput
{
    margin-bottom: 0px;
}

.title, .sub-title
{
    color: #fff;
    text-align: center;
    line-height: 1.2em;
}

.hidden
{
    display: none;
}

@media only screen and (max-width: 470px) {
    
    .container
    {
        padding: 0;
        margin: 0;
        width: 100%;
    }

    .title, .sub-title
    {
        display: none;
    }

    .searchContainer
    {
        width: 100%;
        border-radius: 0px;
        padding: 20px;
        margin: auto;
    }
}