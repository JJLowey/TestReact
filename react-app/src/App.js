import React from 'react';
import axios from 'axios';
import './App.css';

import SearchBar from './Components/SearchBar/SearchBar';

class App extends React.Component
{
    // Set State, defining results as an empty array
    state = { loading: false, results: [] };

    // Called on change of SearchBar component input change
    onSearchChange = async (term) =>
    {
        // If two or more charcaters have been typed
        if(term.length >= 2)
        {
            this.setState({loading: true});

            // Make Get Request using axios.
            const response = await axios.get('https://cors.io/?https://www.rentalcars.com/FTSAutocomplete.do?', 
            {
                // Append required params
                params:
                {
                    solrIndex: "fts_en",
                    solrRows: 6,
                    solrTerm: term
                }
            });

            // Declare array to be populated
            var responseItemArray = [];

            if(response.data.results.numFound > 0)
            {
                // Objective: Return raw data/search results but without any styling
                // Assumption made to display a basic list without any styling
                responseItemArray = response.data.results.docs.map((result, index) => 
                {
                    // Create LI elements with some basic information
                    return (
                        <li key={index}>
                            {(result.placeType === "A" ? "Airport" : (result.placeType === "S") ? "Station" : "City")}
                             - {result.name} {typeof result.iata !== 'undefined' ? '(' + result.iata + ')' : ''}
                        </li>
                    )
                });
            }
            else
            {
                // No results found, return basic string (results.name)
                responseItemArray = <li>No Results Found</li>
            }
            
            // Set the state of the results from async request. 
            // SetState calls render() to show results
            this.setState({loading: false, results: responseItemArray});
        }
    }

    render()
    {
        return (
            <div className="container">
                <h2 className="title col-xs-12">Car Hire â€“ Search, Compare & Save</h2>
                <h5 className="sub-title col-xs-12">Compare 900 companies at over 60,000 locations. Best price guaranteed</h5>
                <div className="searchContainer"> 
                    <h1> Where are you going?</h1>
                    <SearchBar loading={this.state.loading} onChange={this.onSearchChange}/>
                    <ul id="resultsContainer">
                        {this.state.results}
                    </ul>
                </div>
            </div>
        );
    }
}

export default App;