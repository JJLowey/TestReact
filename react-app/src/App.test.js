import React from 'react';
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom';
import App from './App';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';

// Will not recognise setupTests.js :(
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

it('Has not changed?', () => 
{
  // Check if the component has changed? Snapshot is deep, not shallow
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Check App Component Renders', () => 
{
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

describe('Testing App Component', () => 
{
  it('I should see a Search Widget', () => 
  {
    const rootComponent = mount(<App/>)
    expect(rootComponent.find('SearchBar').exists()).toBeTruthy();
  })
})