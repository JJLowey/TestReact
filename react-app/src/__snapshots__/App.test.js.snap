// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Has not changed? 1`] = `
<div
  className="container"
>
  <h2
    className="title col-xs-12"
  >
    Car Hire â€“ Search, Compare & Save
  </h2>
  <h5
    className="sub-title col-xs-12"
  >
    Compare 900 companies at over 60,000 locations. Best price guaranteed
  </h5>
  <div
    className="searchContainer"
  >
    <h1>
       Where are you going?
    </h1>
    <div
      className="col-xs-12"
    >
      <form
        className="col-xs-12"
        onSubmit={[Function]}
      >
        <div
          className="field"
        >
          <label
            htmlFor="pickUpInput"
          >
            Pick-up Location
          </label>
          <input
            autoComplete="off"
            className="form-control col-xs-12 "
            id="pickUpInput"
            onChange={[Function]}
            placeholder="city, airport, station, region and district..."
            type="text"
          />
        </div>
      </form>
    </div>
    <ul
      className=""
      id="resultsContainer"
    />
  </div>
</div>
`;
